src/Host/Client/Impl/Host/IRHostBrokerConnector.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.R.Host.Client.Host {
+    public interface IRHostBrokerConnector : IRHostConnector {
+        /// <summary>
+        /// String token that allows to identify broker that is used to connect to RHost
+        /// For local connections, it is a path to the RHost
+        /// For remote connections, it is a URL
+        /// </summary>
+        string BrokerId { get; }
```

Comment:
Sure. And if we feel like it, we can have a broker connector factory later, that would automatically determine the right connector based on Uri.Schema.

---
In reply to: [73025921](https://github.com/Microsoft/RTVS/pull/2133#discussion_r73025921) [](ancestors = 73025921,73022290)